src/app/services/auth/auth.service.ts

```diff
@@ -172,18 +172,24 @@ export class AuthService {
   /**
    * Checks whether user has any of the supplied roles.
    * Does not ensure that user was loaded.
+   *
+   * Use mockAuth if you need to set user role in tests.
    */
-  hasRole(roles: Role[]): boolean {
-    const currentRoles = this.loggedInUser$?.value?.privilege?.roles?.$set || [];
-    if (!roles?.length || !currentRoles.length) {
-      return false;
-    }
+  hasRole(roles: Role[]): Observable<boolean> {
+    return this.loggedInUser$.pipe(
+      map((user) => {
+        const currentRoles = user?.privilege?.roles?.$set || [];
+        if (!roles?.length || !currentRoles.length) {
+          return false;
+        }
 
-    if (currentRoles.includes(Role.FullAdmin)) {
-      return true;
-    }
+        if (currentRoles.includes(Role.FullAdmin)) {
+          return true;
+        }
 
-    return roles.some((role) => currentRoles.includes(role));
+        return roles.some((role) => currentRoles.includes(role));
```

Comment:
Is there a scenario where all the roles being checked must exist? Not at least one?

Reply:
Not that I'm aware of.